dannybrown/2022/src/day5/day5.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"bufio"
+	"container/list"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type customStack struct {
+    stack *list.List
+}
+
+func (c *customStack) Push(value string) {
+    c.stack.PushFront(value)
+}
+
+func (c *customStack) Pop() error {
+    if c.stack.Len() > 0 {
+        ele := c.stack.Front()
+        c.stack.Remove(ele)
+    }
+    return fmt.Errorf("Pop Error: Stack is empty")
+}
+
+func (c *customStack) Front() (string, error) {
+    if c.stack.Len() > 0 {
+        if val, ok := c.stack.Front().Value.(string); ok {
+            return val, nil
+        }
+        return "", fmt.Errorf("Peep Error: Stack Datatype is incorrect")
+    }
+    return "", fmt.Errorf("Peep Error: Stack is empty")
+}
+
+func (c *customStack) Size() int {
+    return c.stack.Len()
+}
+
+func (c *customStack) Empty() bool {
+    return c.stack.Len() == 0
+}
+
+func check(e error) {
+	if e != nil {
+		panic(e)
+	}
+}
+
+func createStacks() []*customStack {
+	stacks := []*customStack{
+		&customStack{
+			stack: list.New(),
+		},
+		&customStack{
+			stack: list.New(),
+		},
+		&customStack{
+			stack: list.New(),
+		},
+		&customStack{
+			stack: list.New(),
+		},
+		&customStack{
+			stack: list.New(),
+		},
+		&customStack{
+			stack: list.New(),
+		},
+		&customStack{
+			stack: list.New(),
+		},
+		&customStack{
+			stack: list.New(),
+		},
+		&customStack{
+			stack: list.New(),
+		},
+
+	}
+	return stacks
+}
+
+func addRowToStacks(stacks []*customStack, row string) {
+	//                    * [] * haha looks like a face
+	for i := 0; i <= 32; i+=4 {
+		var subsection string
+		if i == 32 {
+			subsection = row[i:]
+		} else {
+			subsection = row[i:i+4]
+		}
+		trimmed := strings.TrimSpace(subsection)
+		if trimmed == "" {
+			continue
+		} else {
+			stacks[i/4].Push(trimmed)
+		}
+	}
+
+}
+
+func main() {
+	dat, err := os.Open("dannybrown/2022/input/day5.txt")
```

Comment:
Oh shit, it's _dat err_.

![dat boi](https://media.giphy.com/media/vc5L6VoTB6tnW/giphy.gif)

Reply:
`addRowToStacks(stacks []*customStack, row string)` aww I forgot to remove the comment when I commited, but the function signature used to be 
`addRowToStacks(stacks *[]*customStack, row string)` until I found out that it is frowned upon/an error to pass a pointer to an array in go. Apparently they can be modified in place which is what I ended up doing